File: conformance/bundles/14-stateless-steps/squares.steps.kt
@file:JvmName("SquaresSteps")

// Kotlin sibling of squares.steps.ts / squares.steps.py / SquaresSteps.java
// (bundle 14-stateless-steps): no state factory — these steps are pure, so
// steps is called without one and handlers run against Unit.
package dev.varar.kotlin.conformance.bundle14

import dev.varar.kotlin.steps
import dev.varar.kotlin.sensor

val steps = steps {
    stimulus("I warm up my mental math") {}
    sensor("The square of {int} is {int}.") { n: Int -> listOf(n, n * n) }
}

File: conformance/bundles/14-stateless-steps/squares.steps.rb
require "varar"

# No initial state: these steps are pure, so steps is called without state and
# handlers get an empty hash.
steps do
  stimulus("I warm up my mental math") { |_state| }

  sensor("The square of {int} is {int}.") { |_state, n, _expected| [n, n * n] }
end
